styx-scheduler-service/src/main/java/com/spotify/styx/state/handlers/ExecutionDescriptionHandler.java

```diff
@@ -61,12 +61,19 @@ public void transitionInto(RunState state) {
     switch (state.state()) {
       case PREPARE:
         try {
-          final Event createdEvent =
-              Event.submit(state.workflowInstance(), getExecDescription(workflowInstance));
-          try {
-            stateManager.receive(createdEvent);
-          } catch (StateManager.IsClosed isClosed) {
-            LOG.warn("Could not send 'created' event", isClosed);
+          final Optional<Workflow> workflow = storage.workflow(workflowInstance.workflowId());
+          if (workflow.isPresent()) {
+            final Event createdEvent = Event.submit(state.workflowInstance(),
+                                                    getExecDescription(workflow.get(), workflowInstance));
+            try {
+              stateManager.receive(createdEvent);
+            } catch (StateManager.IsClosed isClosed) {
+              LOG.warn("Could not send 'created' event", isClosed);
+            }
+          } else {
+            LOG.info(format("Workflow %s does not exist, halting %s", workflowInstance.workflowId(),
+                            workflowInstance));
+            stateManager.receiveIgnoreClosed(Event.halt(workflowInstance));
           }
         } catch (ResourceNotFoundException e) {
           LOG.warn("Failed to prepare execution description for "
```

Comment:
It seems all of the `ResourceNotFoundException`s thrown by getExecDescription are due to user error.

Instead of the above change, how about just changing the logging level here from `warn` to `info`?